lab/src/mapreduce/schedule.go

```diff
@@ -30,5 +30,39 @@ func schedule(jobName string, mapFiles []string, nReduce int, phase jobPhase, re
 	//
 	// Your code here (Part III, Part IV).
 	//
+
+	taskChannel := make(chan int, ntasks)
+	for i := 0; i < ntasks; i++ {
+		go func(taskNumber int) {
+			for {
+				var worker string
+				ok := false
+				taskArgs := DoTaskArgs{
+					JobName:       jobName,
+					File:          mapFiles[taskNumber],
+					Phase:         phase,
+					TaskNumber:    taskNumber,
+					NumOtherPhase: n_other,
+				}
+
+				select {
+				case worker = <-registerChan:
+					ok = call(worker, "Worker.DoTask", taskArgs, nil)
+				}
+
+				if (ok) {
+					taskChannel <- taskNumber
+					registerChan <- worker
+					return
+				}
+			}
+		}(i)
+	}
+
+	// clean up task channel
+	for i := 0; i < ntasks; i++ {
```

Comment:
Hmm, Looks like there is no actual use of taskChannel here.

Reply:
Talked offline. It's used to make the main thread wait until all goroutines finish doing map.